tests/functional/test_install.py

```diff
@@ -228,6 +228,32 @@ def test_install_editable_from_git(script, tmpdir):
     result.assert_installed('pip-test-package', with_files=['.git'])
 
 
+def test_install_editable_uninstalls_existing(data, script, tmpdir):
+    """
+    Test that installing an editable uninstalls a previously installed
+    non-editable version.
+    https://github.com/pypa/pip/issues/1548
+    https://github.com/pypa/pip/pull/1552
+    """
+    to_install = data.packages.join("pip-test-package-0.1.tar.gz")
+    result = script.pip_install_local(to_install)
+    assert 'Successfully installed pip-test-package' in result.stdout
+    result.assert_installed('piptestpackage', editable=False)
+
+    result = script.pip(
+        'install', '-e',
+        '%s#egg=pip-test-package' %
+        local_checkout(
+            'git+http://github.com/pypa/pip-test-package.git',
+            tmpdir.join("cache"),
+        ),
+    )
+    result.assert_installed('pip-test-package', with_files=['.git'])
+    assert 'Found existing installation: pip-test-package 0.1' in result.stdout
+    assert 'Uninstalling pip-test-package:' in result.stdout
+    assert 'Successfully uninstalled pip-test-package' in result.stdout
```

Comment:
also, would be better if you could confirm the package is removed from site-packages using `result.files_deleted`